blockchain/chain_test.go

```diff
@@ -43,9 +43,9 @@ func TestHaveBlock(t *testing.T) {
 	}
 	defer teardownFunc()
 
-	// Since we're not dealing with the real block chain, disable
+	// Since we're not dealing with the real block chain, remove
 	// checkpoints and set the coinbase maturity to 1.
-	chain.DisableCheckpoints(true)
+	chain.RemoveCheckpoints()
```

Comment:
This should be removed as mentioned in my first bullet point in the original review. Allowing the checkpoints to be modified after the fact means the entire checkpoints slice would need to be protected by a mutex. Part of the benefit of passing the checkpoints in an init time is the fact you can then treat them as immutable and then you don't need a mutex.

Instead, they should be set to nil during the initial setup via `chainSetup` (which incidentally they already are because you forgot to modify the call to `blockchain.New` in `common_test.go` to add the field and nil is the zero value for slices).  I would prefer that be made more explicit like so:

```diff
diff --git a/blockchain/common_test.go b/blockchain/common_test.go
index aec85bf..7ba3d7c 100644
--- a/blockchain/common_test.go
+++ b/blockchain/common_test.go
@@ -116,6 +116,7 @@ func chainSetup(dbName string, params *chaincfg.Params) (*blockchain.BlockChain,
        chain, err := blockchain.New(&blockchain.Config{
                DB:          db,
                ChainParams: &paramsCopy,
+               Checkpoints: nil,
                TimeSource:  blockchain.NewMedianTime(),
                SigCache:    txscript.NewSigCache(1000),
        })
```

Then the various callsites of `RemoveCheckpoints` should be removed like so:

```diff
diff --git a/blockchain/chain_test.go b/blockchain/chain_test.go
index 45732d6..860bdbd 100644
--- a/blockchain/chain_test.go
+++ b/blockchain/chain_test.go
@@ -43,9 +43,8 @@ func TestHaveBlock(t *testing.T) {
        }
        defer teardownFunc()

-       // Since we're not dealing with the real block chain, remove
-       // checkpoints and set the coinbase maturity to 1.
-       chain.RemoveCheckpoints()
+       // Since we're not dealing with the real block chain, set the coinbase
+       // maturity to 1.
        chain.TstSetCoinbaseMaturity(1)
```